### PR TITLE
support 24-bit sample types (BDF)

### PR DESCRIPTION
**Pull request by jrevels**
ref #34

still needs tests + a more current maintainer should check behind me on this one

### CODE REVIEW

**Comment by jrevels** [comment]
I've updated this PR to be against #29, since it resolves a bug with that PR

File: src/import_edf.jl [hunk 1]
```diff
@@ -107,14 +107,15 @@ function edf_signal_encoding(edf_signal_header::EDF.SignalHeader,
...
             sample_offset_in_unit=Float64(sample_offset_in_unit),
-            sample_rate=Float64(sample_rate))
+            sample_rate=Float64(sample_rate),
+            sample_type=sample_type)
 end
 
-# TODO: implement a more disciplined widening protocol for `sample_type`
 function promote_encodings(encodings; pick_offset=(_ -> 0.0), pick_resolution=minimum)
-    sample_type = Int16
+    sample_type = reduce(promote_type, (e.sample_type for e in encodings))
```
Comment (ararslan) [inline src/import_edf.jl:118]:
```suggestion
    sample_type = mapreduce(e -> e.sample_type, promote_type, encodings)
```

Comment (jrevels) [reply src/import_edf.jl:118]: AFAICT this is subjective, but I prefer the generator approach/find it more legible

Comment (omus) [inline src/import_edf.jl:118]: I'll call out that `promote_type` takes varargs:
```suggestion
    sample_type = promote_type((e.sample_type for e in encodings)...)
```

**Review by omus — APPROVED** [review]

**Comment by omus** [comment]
> still needs tests

Let's just make an issue for this so these changes to stay in limbo forever

**Comment by jrevels** [comment]
> Let's just make an issue for this so these changes don't stay in limbo forever

a basic test should be simple, i'll add one today

File: src/import_edf.jl (hunk 1 above)
Comment (jrevels) [reply src/import_edf.jl:118]: Yeah, but I prefer not to splat unless i'm in a situation where the splatted element count is known locally (here it's not - it's caller driven)

**Comment by jrevels** [comment]
given the approval i'll go ahead and merge this in

**Pull request merged**